azure-pipelines.yml

```diff
@@ -0,0 +1,761 @@
+      azureConnectionType: 'ConnectedServiceNameARM'
+      azureSubscription: 'SDKInfrastructureSP'
+      destination: AzureBlob
+      storage: azureclistage
+      containerName: archive-devops
+      blobPrefix: '$(Build.BuildNumber)'
+  - task: ArchiveFiles@2
+    displayName: 'Create tarball of source'
+    inputs:
+      rootFolderOrFile: '$(Build.SourcesDirectory)'
+      archiveType: tar
+      tarCompression: gz
+      archiveFile: '$(Build.ArtifactStagingDirectory)/azure-cli_$(Build.BuildId).tar.gz'
+  - task: AzureFileCopy@1
+    displayName: 'AzureBlob File Copy'
+    inputs:
+      sourcePath: '$(Build.ArtifactStagingDirectory)/azure-cli_$(Build.BuildId).tar.gz'
+      azureConnectionType: 'ConnectedServiceNameARM'
+      azureSubscription: 'SDKInfrastructureSP'
+      destination: AzureBlob
+      storage: azureclistage
+      containerName: archive-devops
+      blobPrefix: '$(Build.BuildNumber)'
+
+
+- phase: Phase_11
+  displayName: Build Windows MSI
+
+  dependsOn: Phase_13
+  condition: succeeded()
+  queue:
+    name: Hosted VS2017
+  steps:
+  - task: DownloadBuildArtifacts@0
+    displayName: 'Download Build Artifacts'
+    inputs:
+      artifactName: metadata
+
+
+  - script: |
+       set /p CLI_VERSION=<$(System.ArtifactsDirectory)/metadata/version
+       set
+
+       build_scripts/windows/scripts/build.cmd
+    displayName: 'Build Windows MSI'
+
+  - task: PublishBuildArtifacts@1
+    displayName: 'Publish Artifact: MSI'
+    inputs:
+      PathtoPublish: 'build_scripts/windows/out/'
+
+      ArtifactName: msi
+
+
+
+- phase: Phase_4
+  displayName: Build Docker Image
+
+  dependsOn: Phase_13
+  condition: succeeded()
+  queue:
+    name: Hosted Ubuntu 1604
+  steps:
+  - task: Bash@3
+    displayName: 'Bash Script'
+    inputs:
+      targetType: 'filePath'
+      filePath: scripts/release/docker/pipeline.sh
+
+
+  - task: PublishBuildArtifacts@1
+    displayName: 'Publish Artifact: docker image'
+    inputs:
+      ArtifactName: docker
+
+
+
+- phase: Phase_19
+  displayName: Test Docker Image
+
+  dependsOn: Phase_4
+  condition: succeeded()
+  queue:
+    name: Hosted Ubuntu 1604
+  steps:
+  - task: DownloadBuildArtifacts@0
+    displayName: 'Download Metadata'
+    inputs:
+      artifactName: metadata
+
+
+  - task: DownloadBuildArtifacts@0
+    displayName: 'Download Docker Image'
+    inputs:
+      artifactName: docker
+
+
+  - bash: |
+       set -exv
+
+       CLI_VERSION=`cat $SYSTEM_ARTIFACTSDIRECTORY/metadata/version`
+       IMAGE_NAME=clibuild$BUILD_BUILDNUMBER:latest
+       TAR_FILE=$SYSTEM_ARTIFACTSDIRECTORY/docker/docker-azure-cli-$CLI_VERSION.tar
+
+       echo "== Test docker image =="
+
+       docker load < $TAR_FILE
+       docker run $IMAGE_NAME /bin/bash -c "time az self-test && time az --version && sleep 5"
+    displayName: 'Bash Script'
+
+
+- phase: Phase_1
+  displayName: Build Python Wheels
+
+  dependsOn: Phase_13
```

Comment:
Phase_1 depends on Phase_13?  While these were probably autogenerated names, we should try to make them reflect reality. I think this file would already be confusing to most people. Things like this make it more confusing than it needs to be.

Reply:
Fixed in d7ba697